arangod/Utils/CollectionNameResolver.cpp

```diff
@@ -292,7 +292,9 @@ std::string CollectionNameResolver::localNameLookup(TRI_voc_cid_t cid) const {
         std::shared_ptr<LogicalCollection> ci;
         try {
           ci = ClusterInfo::instance()->getCollection(
-              (*it).second->dbName(), name);
+            std::dynamic_pointer_cast<LogicalCollection>(it->second)->dbName(),
```

Comment:
please use check:
```
it->second->category() == LogicalCollection::category()
```
and
```
std::static_pointer_cast<LogicalCollection>(it->second)
```
instead of
```
std::dynamic_pointer_cast<LogicalCollection>(it->second)
```
since latter may cause `segfault`

Reply:
CollectionNameResolver related changes will be done in the next part of task #344